### ISSUE TITLE
Please build for Scala 2.12.0

### ISSUE THREAD

**Issue by countfloyd** [altoo-ag/akka-kryo-serialization#105]

**Comment by guidomedina** [comment]
@romix if it helps, this [commit](https://github.com/romix/akka-kryo-serialization/commit/f00ec6889db191b6f81a5a967e0c81291113b4ea#diff-215113124f1de02f228327ba7abb45f7) is the one where you removed the cross scala versions because of incompatibility with Scala 2.10.x.

I think you can re-add it but for only 2.11.x and 2.12.x as both are now stable.

**Comment by guidomedina** [comment]
@romix I have tried to make a PR to help but I ran into some small issues, I'm no Scala developer so it is more difficult for me to fix them:
- `-optimise` option has been rewritten for Scala 2.12.0
- 1 test is failing.
- some deprecations.

Also in a console do an `sbt clean compile` and see the warnings.

You can use my commit as base as start point if you wish:
https://github.com/guidomedina/akka-kryo-serialization/commit/0379023fb878d2270db1dae80420ca7b6b829b4f

**Comment by TrustNoOne** [comment]
If you change the failing test like this
```scala
  it should "roundtrip immutable LongMap" in {
     ...
    kryo.register(classOf[scala.collection.immutable.LongMap[_]], 3042) // instead of LongMap[Any]
```

it passes, but I have no Idea why

**Comment by guidomedina** [comment]
Thanks @TrustNoOne 
That's one down, we only need @romix to jump in and figure out the other small issues, I'm sure for a Scala developer these small issues are easy to fix.

**Comment by TrustNoOne** [comment]
umh, it looks like it's failing again but I don't understand why, It passed once when I ran it in intelliJ O_o

**Comment by luben** [comment]
I will send a PR soon, I have it working but want to clean it up.

**Comment by briantopping** [comment]
Are there any bug fixes in 0.5.1 or 0.5.2 that 2.11 users would want? Because both of these versions are currently missing in Central. Thanks! :)

**Comment by spaced** [comment]
v0.5.1 is build against Akka-2.4 and Kryo-4.0 and is available for Scala-2.11 and Scala-2.12
